src/components/navigation/document-browser-scroller.tsx

```diff
@@ -0,0 +1,117 @@
+import { ISubTabSpec, NavTabModelType } from "../../models/view/nav-tabs";
+import { DocumentCollectionList } from "../thumbnail/document-collection-list";
+import ScrollArrowIcon from "../../assets/scroll-arrow-icon.svg";
+import CollapseScrollerIcon from "../../assets/show-hide-document-view-icon.svg";
+
+import "./document-browser-scroller.scss";
+
+interface DocumentBrowserScrollerProps {
+  subTab: ISubTabSpec;
+  tabSpec: NavTabModelType;
+  openDocumentKey: string;
+  openSecondaryDocumentKey: string;
+  onSelectDocument: (document: DocumentModelType) => void;
+}
+
+export const DocumentBrowserScroller =
+    ({subTab, tabSpec, openDocumentKey, openSecondaryDocumentKey, onSelectDocument}: DocumentBrowserScrollerProps) => {
+  const [scrollerCollapsed, setScrollerCollapsed] = useState(false);
+  const [collectionElement, setCollectionElement] = useState<HTMLDivElement>();
+  const documentScrollerRef = useRef<HTMLDivElement>(null);
+  const [scrollToLocation, setScrollToLocation] = useState(0);
+  const [panelWidth, setPanelWidth] = useState(0);
+
+  const scrollWidth = collectionElement?.scrollWidth ?? 0;
+
+  useEffect(() => {
+    if(scrollToLocation !== undefined) {
+      collectionElement?.scrollTo({left: scrollToLocation, behavior: "smooth"});
+    }
+  },[collectionElement, scrollToLocation]);
+
+  // Keep track of the size of the containing element
+  useEffect(() => {
+    let obs: ResizeObserver;
+    if (documentScrollerRef.current) {
+      obs = new ResizeObserver(() => {
+        setPanelWidth(documentScrollerRef.current?.clientWidth ?? 0);
+      });
+      obs.observe(documentScrollerRef.current);
+    }
+
+    return () => obs?.disconnect();
+  }, []);
+
+  const handleScrollTo = (side: string) => {
+    const direction = side === "left" ? -1 : 1;
+    const attemptedScrollTo = scrollToLocation + direction * panelWidth;
+    const scrollTo = Math.max(0, Math.min(scrollWidth - panelWidth, attemptedScrollTo));
+    setScrollToLocation(scrollTo);
+  };
+
+  const handleCollapseScroller = () => {
+    setScrollerCollapsed(!scrollerCollapsed);
+  };
+
+  return (
+    <>
+      <div className={classNames("document-scroller", tabSpec.tab, {"collapsed": scrollerCollapsed})}
+          ref={documentScrollerRef}>
+        <DocumentCollectionList
+            setCollectionElement={setCollectionElement}
+            subTab={subTab}
+            tabSpec={tabSpec}
+            horizontal={true}
+            collapsed={scrollerCollapsed}
+            selectedDocument={openDocumentKey}
+            selectedSecondaryDocument={openSecondaryDocumentKey}
+            scrollToLocation={scrollToLocation}
+            onSelectDocument={onSelectDocument}
+        />
+        {(scrollToLocation > 0) && !scrollerCollapsed &&
+            <ScrollEndControl side={"left"} tab={tabSpec.tab} onScroll={handleScrollTo} />
+        }
+        {(scrollToLocation < scrollWidth - panelWidth) && !scrollerCollapsed &&
```

Comment:
I'd use a constant for `scrollWidth - panelWidth` since it's used multiple times (also on line 51).